mobile/src/main/java/org/openhab/habdroid/util/Util.java

```diff
@@ -185,6 +187,43 @@ public static void setActivityTheme(@NonNull final Activity activity, String the
         return R.style.HABDroid_Light;
     }
 
+    /**
+     * If fullscreen is enabled and we are on at least android 4.4 set
+     * the system visibility to fullscreen + immersive + noNav
+     *
+     * @author Dan Cunningham
+     */
+    public static void checkFullscreen(Activity activity) {
+        checkFullscreen(activity, isFullscreenEnabled(activity));
+    }
+
+    public static void checkFullscreen(Activity activity, boolean isEnabled) {
+        int uiOptions = activity.getWindow().getDecorView().getSystemUiVisibility();
+        if (isEnabled) {
+            uiOptions = uiOptions
+                    | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                    | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+                    | View.SYSTEM_UI_FLAG_FULLSCREEN;
+        } else {
+            uiOptions = uiOptions
+                    & View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                    & View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+                    & View.SYSTEM_UI_FLAG_FULLSCREEN;
+        }
+        activity.getWindow().getDecorView().setSystemUiVisibility(uiOptions);
+    }
+
+    /**
+     * If we are 4.4 we can use fullscreen mode and Daydream features
+     */
```

Comment:
Comment seems off here. Maybe use `// The system UI flags for fullscreen mode only work on 4.4+`